LibAddonMenu-2.0/LibAddonMenu-2.0/controls/editbox.lua

```diff
@@ -15,12 +17,28 @@
 } ]]
 
 
-local widgetVersion = 14
+local widgetVersion = 15
 local LAM = LibStub("LibAddonMenu-2.0")
 if not LAM:RegisterWidget("editbox", widgetVersion) then return end
 
 local wm = WINDOW_MANAGER
 
+local function GetValidTextType(textType)
+    textType = LAM.util.GetDefaultValue(textType)
+    if textType and (textType >= TEXT_TYPE_ITERATION_BEGIN and textType <= TEXT_TYPE_ITERATION_END) then
+        return textType
+    end
+    return TEXT_TYPE_ALL
+end
+
+local function GetValidMaxChars(number)
+    number = LAM.util.GetDefaultValue(number)
+    if number and type(number) == "number" then
+        return number
+    end
+    return MAX_GUILD_RECRUITMENT_MESSAGE_LENGTH

```

Comment:
Isn't MAX_GUILD_RECRUITMENT_MESSAGE_LENGTH less than the previous default of 3000? Either way it's not good to change the default value as it may break expectations.

Reply:
good point, changed it back

Reply:
fixed. maybe we could add an additional check:
`if type(number) == "number" and number > 0 then`

Reply:
Don't think that's needed. SetMaxInputChars should already check if it can handle the value internally and act accordingly? Actually you could just drop the type check and just pass it like so:
```
     editbox:SetMaxInputChars(editboxData.maxChars or 3000) 
```
That way the dev should get an error in the LDL log showing that they provided a wrong value type instead of having to guess why maxChars doesn't get applied.

Reply:
yeah just tested. 0 will automatically turns into 1, a string throws an error

Reply:
And careful with my suggestion. I forgot the call to GetDefaultValue ;)

Reply:
no i didn't forget it, i just thought it makes sense to use it for the textType, but not really for the maxChars usecase. but i can add it if you want

Reply:
I'm not sure if it makes sense, but maybe someone will use it some day, so it wouldn't hurt to add it.

Reply:
done